### PR TITLE
tidbmonitor support dynamic reload configuration

### LINKED ISSUE #4020: Dynamic update tidbmonitor when prometheus configuration update.
## Feature Request

**Is your feature request related to a problem? Please describe:**
Dynamic update tidbmonitor when prometheus or grafana configuration changed.

**Describe the feature you'd like:**
I think we can add a field called `revision` into `spec` and `status`.
When configuration changed, we can generate new revision hash value and set it to `spec` revision field. The tidbmonitor  find the status and spec field value not equal, it can trigger sts recreating.

### PR DESCRIPTION
**Pull request by mikechengwei**
### What problem does this PR solve?
fix #4020
### What is changed and how does it work?
Use prometheus-config-reloader dynamic update configuration function.

### CODE REVIEW

**Comment by mikechengwei** [comment]
YAML:
```
apiVersion: pingcap.com/v1alpha1
kind: TidbMonitor
metadata:
  name: monitor
spec:
  clusterScoped: true
  clusters:
    - name: ns1
      namespace: ns1
    - name: ns2
      namespace: ns2
  prometheus:
    baseImage: prom/prometheus
    version: v2.11.1
  grafana:
    baseImage: grafana/grafana
    version: 7.5.7
  initializer:
    baseImage: pingcap/tidb-monitor-initializer
    version: v5.1.1
  reloader:
    baseImage: quay.io/prometheus-operator/prometheus-config-reloader
    version: v0.49.0
  imagePullPolicy: IfNotPresent
```

**Comment by mikechengwei** [comment]
Manual Test screenshots:
image

File: pkg/monitor/monitor/util.go [hunk 1]
```diff
@@ -440,7 +440,7 @@ func getMonitorPrometheusContainer(monitor *v1alpha1.TidbMonitor, tc *v1alpha1.T
 	} else {
 		retention = fmt.Sprintf("%dd", monitor.Spec.Prometheus.ReserveDays)
 	}
-	commands := []string{"sed 's/$NAMESPACE/'\"$NAMESPACE\"'/g;s/$POD_NAME/'\"$POD_NAME\"'/g' /etc/prometheus/config/prometheus.yml > /etc/prometheus/config_out/prometheus.yml && /bin/prometheus --web.enable-admin-api --web.enable-lifecycle --config.file=/etc/prometheus/config_out/prometheus.yml --storage.tsdb.path=/data/prometheus --storage.tsdb.retention.time=" + retention}
+	commands := []string{"sed -e '5s/(//g' -e '5s/)//g' -e 's/$NAMESPACE/'\"$NAMESPACE\"'/g;s/$POD_NAME/'\"$POD_NAME\"'/g' /etc/prometheus/config/prometheus.yml > /etc/prometheus/config_out/prometheus.yml && /bin/prometheus --web.enable-admin-api --web.enable-lifecycle --config.file=/etc/prometheus/config_out/prometheus.yml --storage.tsdb.path=/data/prometheus --storage.tsdb.retention.time=" + retention}
```
Comment (DanielZhangQD) [inline pkg/monitor/monitor/util.go:443]: Can we combine the sed commands?

Comment (mikechengwei) [reply pkg/monitor/monitor/util.go]: sed -e '5s/[()]//g'

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Review by csuzhangxc — APPROVED** [review]

**Pull request merged**